bip-0174.mediawiki

```diff
@@ -494,6 +494,39 @@ The currently defined per-input types are defined as follows:
 | 0, 2
 | [[bip-0371.mediawiki|371]]
 |-
+| ECDSA P2C Key Tweak
+| <tt>PSBT_IN_ECDSA_P2C_TWEAK = 0x19</tt>
+| <tt><pubkey></tt>
+| 33 bytes of compact public key serialization specifying to which of keys the ECDSA tweak must be applied (i.e. this MUST be a value of a public key before the tweak is applied)
+| <tt><tweak></tt>
+| The 32 byte value which MUST be added to a private key to produce correct ECDSA signature ("key tweak"). Signers should remove this field after <tt>PSBT_IN_PARTIAL_SIG</tt> is constructed.
+|
+|
+| 0, 2
+| 174
+|-
+| Taproot Per-key P2C Tweak
+| <tt>PSBT_IN_TAP_P2C_TWEAK = 0x1a</tt>
+| <tt><xonlypubkey></tt>
+| 32 bytes of public key specifying to which of keys tweak must be applied (i.e. this MUST be a value of a public key before the tweak is applied)
```

Comment:
Does there need to be any consideration for keys in leaf scripts?

Reply:
I expect this to be a bad practice: the commitment must be singular per output (for instance to prevent double spends in second-layer systems like RGB), so allowing multiple tweaks in the same output is a security risk.

Reply:
I would like to support keys in leaf scripts (and arbitrary numbers of tweaks on arbitrary sets of keys). If the possibility is a security risk for some other protocol there is nothing that PSBT can do about it.

Reply:
Ok, probably I was overcautious.

Reply:
@apoelstra what do you mean by "tweaks on arbitrary sets of keys":
- the same tweak must be applied to each of the keys in the set;
- the tweak must be applied to one and only one key (i.e. _the sum_ of this keys commits to the tweak)?

Reply:
I came to conclusion that even most complex cases can be covered by a quite simple setup - pls see my description in https://lists.linuxfoundation.org/pipermail/bitcoin-dev/2022-January/019761.html